### PR TITLE
#168781624 -  Social Media Login

### PR DESCRIPTION
**Pull request by Cheza-Dzabala**
#### What does this PR do?

Allows a user to authenticate their account using Facebook & Google

#### Description of Task to be completed?
- A user should be able to login by Facebook
- A user should be able to login by Google

#### How should this be manually tested?
- Clone this Repo
- Run npm install
- Run npm run dev
- Access localhost:8080/users/login
- Select Either Facebook or Google
- Note authentication

#### Any background context you want to provide?
N / A
#### Screenshots (if appropriate)
Screen Shot 2019-12-09 at 12 32 31

Screen Shot 2019-12-09 at 08 05 11
Screen Shot 2019-12-09 at 08 05 29

#### Questions:
N / A

### CODE REVIEW

**Review by alainmateso — CHANGES_REQUESTED** [review]
Great Job Cheza. Kindly revise your Google Login. It seems like the backend app breaks when you try to login with google.

**Review by sabin18 — APPROVED** [review]
@Cheza-Dzabala  good job!

File: src/actions/authActions.js [hunk 1]
```diff
@@ -15,3 +17,9 @@ export const signupAction = (userData) => (dispatch) => backendCall.post('/users
   .catch((err) => {
     dispatch(authType(SIGNUP_FAIL, err.response.data));
   });
+
+export const saveUserToState = (userData) => (dispatch) => {
+  window.localStorage.setItem('token', userData.token);
+  dispatch(authType(SOCIAL_AUTH_PENDING, true));
+  dispatch(authType(SOCIAL_AUTH_SUCCESS, userData));
+};
```
Comment (nakiwuge) [inline src/actions/authActions.js:32]: Give this function a more meaningful name. Also remove the pending type and move it to the component by creating an `isLoading` state
Add a catch too

File: src/actions/authActions.js [hunk 2]
```diff
@@ -15,3 +17,9 @@ export const signupAction = (userData) => (dispatch) => backendCall.post('/users
   .catch((err) => {
     dispatch(authType(SIGNUP_FAIL, err.response.data));
   });
+
+export const saveUserToState = (userData) => (dispatch) => {
+  window.localStorage.setItem('token', userData.token);
```
Comment (nakiwuge) [inline src/actions/authActions.js]: you can use localStorage.setItem()

File: src/components/Register.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import SocialButtons from './pages/SocialButtons';
+
+export default function Register() {
+  return (
+    <div>
+      <h1>Register</h1>
+      <SocialButtons />
+    </div>
+  );
+}
```
Comment (nakiwuge) [inline src/components/Register.js]: remove this file

File: src/components/pages/SocialButtons.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    {text}
+  </a>
+);
+
+const SocialButtons = () => (
+  <div className="button-holder">
+    {buttons.map((btn) => renderButton(btn.class, btn.href, btn.logo, btn.key, btn.text))}
+  </div>
+);
+
```
Comment (nakiwuge) [inline src/components/pages/SocialButtons.js:23]: reuse the button component

File: src/tests/actionsTests/SocialAuthActionTest.spec.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { SOCIAL_AUTH_PENDING } from '../../actions/types';
+import store from '../../reduxStore';
+
+describe('Redux', () => {
```
Comment (nakiwuge) [inline src/tests/actionsTests/SocialAuthActionTest.spec.js]: Give a describe a more descriptive test

File: src/views/AuthSuccess.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+/* eslint-disable import/prefer-default-export */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import extractUserFromQuery from '../utilities/objectExtraction';
+import { saveUserToState } from '../actions/authActions';
+
+const AuthSuccess = (props) => {
```
Comment (nakiwuge) [inline src/views/AuthSuccess.js]: Rename this function to `SocialAuth`

File: src/views/AuthSuccess.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  const { location, saveUserToState, status } = props;
+  const { search } = location;
+  const user = extractUserFromQuery(search);
+  saveUserToState(user);
```
Comment (nakiwuge) [inline src/views/AuthSuccess.js]: Put this in `componentDidMount()`

**Review by alainmateso — APPROVED** [review]

**Pull request merged**